autotest/test_modeltime.py

```diff
@@ -0,0 +1,292 @@
+import datetime
+
+import numpy as np
+import pandas as pd
+
+import flopy
+from flopy.discretization.modeltime import ModelTime
+
+
+def test_date_userinput_parsing():
+    formats = [
+        datetime.datetime(2024, 11, 12),
+        np.datetime64("2024-11-12"),
+        pd.Timestamp("2024-11-12"),
+        "2024-11-12",
+        "2024/11/12",
+        "11-12-2024",
+        "11/12/2024",
+    ]
+
+    valid = datetime.datetime(2024, 11, 12)
+
+    for dt_rep in formats:
+        dt_obj = ModelTime.parse_datetime(dt_rep)
+        if dt_obj != valid:
+            raise AssertionError("datetime not properly determined from user input")
+
+
+def test_datetime_userinput_parsing():
+    formats = [
+        datetime.datetime(2024, 11, 12, 14, 31, 29),
+        np.datetime64("2024-11-12T14:31:29"),
+        pd.Timestamp("2024-11-12T14:31:29"),
+        "2024-11-12T14:31:29",
+        "2024/11/12T14:31:29",
+        "11-12-2024 14:31:29",
+        "11/12/2024 14:31:29",
+    ]
+
+    valid = datetime.datetime(2024, 11, 12, 14, 31, 29)
+
+    for dt_rep in formats:
+        dt_obj = ModelTime.parse_datetime(dt_rep)
+        if dt_obj != valid:
+            raise AssertionError("datetime not properly determined from user input")
+
+
+def test_timeunits_userinput_parsing():
```

Comment:
consider refactoring test cases like this (here and in the functions below) to use `@pytest.mark.parametrize`. For example, this test could be refactored to:
```python
@pytest.mark.parametrize('unit_name,checks', (
        ("years", ["years", "YeaR", "yaEr", "ayer", "y", "yr", 5]),
        ("days", ["days", "Day", "dyAs", "dysa", "d", 4]),
        ("hours", ["hours", "Hour", "huors", "h", "hrs", 3]),
        ("minutes", ["minutes", "MinUte", "minte", "m", "min", 2]),
        ("seconds", ["seconds", "Second", "sedcon", "s", "sec", 1]),
        ("unknown", ["unkonwn", "undefined", "u", 0])
        ))
def test_timeunits_userinput_parsing(unit_name,checks):
    for check in checks:
        mt_unit = ModelTime.timeunits_from_user_input(check)
        if mt_unit != unit_name:
            raise AssertionError(
                "Units are unable to be determined from user input"
            )
```
This carries some advantages, in that each case is treated as a separate test (meaning all of the cases will be run unless "fail fast" is set), and the tests can be debugged individually in VS Code.